### ISSUE TITLE
Feature Request: Code formatters

### ISSUE THREAD

**Issue by bernd-wechner** [linuxmint/xed#139]
Perhaps in the realm of a plugin or extension, but I use xed a lot now (being the default on Mint) and it's nice but a need I often have and would love to see on my local editor is reformatting code.  Now before you jump in and think "can of worms" and "huge topic" (which I'd concur with hence am forestalling) I'm thinking of the basics that I see in some slightly more advanced editors, namely HTML/XML, JSON and even Javascript as basic text types that I'm often wanting formatted because I nab them form some place and they a munged and compressed.

Plenty of websites do this and I tend to use them instead. But a local solution is much slicker and I suspect a hook for third party reformatters via a plugin is not a big deal (and then finding those formatters). 

Just a thought. Would make xed that little bit more awesome.

**Comment by mathieu-rossignol** [comment]
+1. That's a great missing feature of this great editor coming with mint.

**Comment by JamesTheAwesomeDude** [comment]
Just to confirm: the Xed editor still does not have a way to "pretty-print" a JSON file?

**Comment by bernd-wechner** [comment]
I haven't noticed it. I have been using Atom for this in the interim. Atom hase nice beautifier plugins and my most common needs in the interim have been JS, JSON, SQL and HTML and I can paste any of these into Atom and CTL+ALT+B sees them beautified (reformatted elegantly).

Still, this request stands. As xed is the default editor, I would see such beautification of very common languages (of which JS, JSON, SQL and HTML) are examples a huge asset. With teh experience of Atom behind me 4 years on I can say though that:

1. There are CLI tools that do this formatting already and an editor only needs to outsource the job to them, not reimplement it. 
2. The tool to use must be preconfigured or configurable based on file type and file type can be inferred from file extension but a UI mechanism for specifying the file type of an unsaved file (a past into a new editor tab as is common for me) is needed as a consequence. Atom offers this through a spot on the window's status bar (bottom border hugging panel of the window) where the file type is displayed if you click it you can select from the list of known types).

In short though this is not a huge code extension to a basic editor, a type to tool mapping config, and type detection and setting feature but IMHO a pretty hefty value add, as beautifying is such a common need. I mean Atom has plugins for a squidzilllion formats and sometimes I find more esoteric ones useful too, like bash or ini files etc. Rarely I admit though, as the more complicated a language/format is the more iffy beautification becomes and beautifiers often do stuff I don't want or have a squidzillion configs to tune to do what I want - I beautify Python for example on save in PyDev - does it automatically, but I had to go through the configs to set it up so it doesn't do things I don't want ;-).
